### PR TITLE
Problem: IMDSv1 used to obtain the credentials information

### PR DESCRIPTION
**Pull request by linfeng-crypto**
get credentials from Instance Metadata Service Version 2

### CODE REVIEW

File: providers/nitro/nitro-helper/src/key_utils.rs [hunk 1]
```diff
@@ -39,21 +43,57 @@ pub(crate) mod credential {
...
-    /// Gets the role name to get credentials for using the IAM Metadata Service (169.254.169.254).
+    /// get credentials from Aws Instance Metadata Service Version 2
+    /// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
     pub fn get_credentials() -> Result<AwsCredentials, String> {
+        let token_address = format!("http://{}/{}/", AWS_CREDENTIALS_PROVIDER_IP, AWS_TOKEN_PATH);
+        let client = Client::new();
+        let header_key = HeaderName::from_str("X-aws-ec2-metadata-token-ttl-seconds")
+            .map_err(|_| "invalid header name".to_string())?;
+        let header_value =
+            HeaderValue::from_str("10").map_err(|_e| "invalid header value".to_string())?;
```
Comment (tomtau) [inline providers/nitro/nitro-helper/src/key_utils.rs:54]: isn't 10 seconds too short?

File: providers/nitro/nitro-helper/src/key_utils.rs [hunk 2]
```diff
@@ -39,21 +43,57 @@ pub(crate) mod credential {
         }
     }
 
-    /// Gets the role name to get credentials for using the IAM Metadata Service (169.254.169.254).
+    /// get credentials from Aws Instance Metadata Service Version 2
+    /// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
     pub fn get_credentials() -> Result<AwsCredentials, String> {
+        let token_address = format!("http://{}/{}/", AWS_CREDENTIALS_PROVIDER_IP, AWS_TOKEN_PATH);
+        let client = Client::new();
+        let header_key = HeaderName::from_str("X-aws-ec2-metadata-token-ttl-seconds")
```
Comment (tomtau) [inline providers/nitro/nitro-helper/src/key_utils.rs:51]: would error every be returned in `HeaderName::from_str` for this constant?

**Review by tomtau — CHANGES_REQUESTED** [review]
perhaps the session token expiration can be made configurable or extended to some more reliable time (a few minutes?)?

File: providers/nitro/nitro-helper/src/key_utils.rs (hunk 1 above)
Comment (linfeng-crypto) [reply providers/nitro/nitro-helper/src/key_utils.rs:54]: each time we request a new token and then use it to get the credentials, 10s is enough or we can change to 30s.

File: providers/nitro/nitro-helper/src/key_utils.rs (hunk 2 above)
Comment (linfeng-crypto) [reply providers/nitro/nitro-helper/src/key_utils.rs:51]: no, before I use `unwrap`, I'll return it back.

**Review by tomtau — APPROVED** [review]

**Pull request merged**